lib/parser/json_parser.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const _ = require('lodash'),
+    JDLObject = require('../core/jdl_object'),
+    JDLEntity = require('../core/jdl_entity'),
+    JDLField = require('../core/jdl_field'),
+    JDLValidation = require('../core/jdl_validation'),
+    JDLEnum = require('../core/jdl_enum'),
+    JDLRelationship = require('../core/jdl_relationship'),
+    JDLRelationships = require('../core/jdl_relationships'),
+    JDLUnaryOption = require('../core/jdl_unary_option'),
+    JDLBinaryOption = require('../core/jdl_binary_option'),
+    Validations = require('../core/jhipster/validations'),
+    RelationshipTypes = require('../core/jhipster/relationship_types').RELATIONSHIP_TYPES,
+    UnaryOptions = require('../core/jhipster/unary_options').UNARY_OPTIONS,
+    BinaryOptions = require('../core/jhipster/binary_options').BINARY_OPTIONS;
+
+module.exports = {
+  parseEntities: parseEntities,
+  parseServerOptions: parseServerOptions
+}
+
+/*
+ * Parses the json entities into JDL
+ * entities: json map with entity name as key and entity definition as value
+ * jdl: JDLObject to which the parsed entities are added. If undefined a new JDLObject is created.
+ * returns the JDLObject
+ */
+function parseEntities(entities, jdl) {
+  if (!entities) {
+    throw new buildException(
+      exceptions.NullPointer,
+      'Entities have to be passed to be converted.');
+  }
+  if (jdl === undefined) {
```

Comment:
If you pass null, you're voluntarily asking for trouble and you'll get a NPE quite quickly! Throwing a buildException would be cleaner I guess.